src/pages/instances/InstanceSearchFilter.tsx

```diff
@@ -56,24 +53,74 @@ const InstanceSearchFilter: FC<Props> = ({ instances, setFilters }) => {
       }),
     },
   ];
-
+  const [_, setSearchParams] = useSearchParams();
   const onSearchDataChange = (searchData: SearchAndFilterChip[]) => {
-    setFilters({
-      queries: searchData
-        .filter((chip) => chip.quoteValue)
-        .map((chip) => chip.value.toLowerCase()),
-      statuses: enrichStatuses(
-        searchData
-          .filter((chip) => chip.lead === "status")
-          .map((chip) => chip.value as LxdInstanceStatus),
-      ),
-      types: searchData
-        .filter((chip) => chip.lead === "type")
-        .map((chip) => (chip.value === "VM" ? "virtual-machine" : "container")),
-      profileQueries: searchData
-        .filter((chip) => chip.lead === "profile")
-        .map((chip) => chip.value),
+    const filteredQueries = searchData
+      .filter((chip) => chip.quoteValue === true || chip.lead === "query")
+      .map((chip) => chip.value.toLowerCase());
+
+    const filteredStatuses = searchData
+      .filter((chip) => chip.lead === "status")
+      .map((chip) => chip.value as LxdInstanceStatus);
+    const filteredTypes = searchData
+      .filter((chip) => chip.lead === "type")
+      .map((chip) => (chip.value === "VM" ? "virtual-machine" : "container"));
+
+    const profileQueries = searchData
+      .filter((chip) => chip.lead === "profile")
+      .map((chip) => chip.value);
+
+    const url = new URL(window.location.origin + window.location.pathname);
+
+    filteredQueries.forEach((query) => {
+      url.searchParams.append("queries", query);
+    });
+
+    filteredStatuses.forEach((status) => {
+      url.searchParams.append("statuses", status);
     });
+
+    filteredTypes.forEach((type) => {
+      url.searchParams.append("types", type);
+    });
+    profileQueries.forEach((profile) => {
+      url.searchParams.append("profileQueries", profile);
```

Comment:
See previous comment - if you decide to simplify the query params as suggested, these lines and the ones a bit below should also be updated.